front/app/components/EsriMap/index.tsx

```diff
@@ -234,6 +231,15 @@ const EsriMap = ({
     }
   }, [onHover, mapView]);
 
+  useEffect(() => {
+    // Once mapView is created, run onInit function if it was provided
+    if (onInit) {
+      if (mapView) {
+        onInit(mapView);
+      }
+    }
+  }, [mapView, onInit]);
```

Comment:
Hmm is it required that `onInit` is only called once? Because this way that's not guaranteed- it will be called every time `onInit` changes. This will depend on the dependencies it has in the parent. With the previous solution this was not the case.

If you don't want to have it in `initialData` anymore, you could do something like `const [initialized, setInitialized] = useState(false)` and then only call `onInit` in this `useEffect` if `initialized === false`, and after calling it `setInitialized(true)`

Reply:
Good point! I'll change that :+1: It wasn't working correctly when I had it in the initialData, but maybe one of the other changes has now fixed that. I'll try putting it back in initialData, and if it doesn't fix it I'll implement your suggestion.

Thanks Luuc!

Reply:
Fixed! Added it back into the initialData :+1: